### PR TITLE
First version of user documentation.

### PR DESCRIPTION
**Pull request by PicoCentauri**
Based on the work by @curiosity54 and @HowWeiBin this PR adds a first version of a user documentation. The structure is similar to rascalines but of course with different content.

If you review take a close look at the general structure since the content is in a rough state.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    flake8 {toxinidir}/python {toxinidir}/setup.py
+    black --check --diff {toxinidir}/python {toxinidir}/setup.py
+    isort --check-only --diff {toxinidir}/python {toxinidir}/setup.py
+
+[testenv:docs]
+deps =
+    {[testenv]deps}
+    -r docs/requirements.txt
+commands =
+    pip install . --quiet
```
Comment (Luthaf) [inline tox.ini:40]: This should be something like 

```
bash -c "rm -rf dist && python setup.py --quiet bdist_wheel && pip install --force-reinstall --no-deps ./dist/equistore-*.whl"
```

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Equistore is a specialized data storage format suited to all your
+atomistic simulation needs and more.
+
+The core of the library is written in Rust and we provide
+APIs for C/C++ and Python as well.
+
+.. warning::
+
+    **Rascaline is still as the proof of concept stage. You should not use it for
+    anything important.**
```
Comment (Luthaf) [inline README.rst:15]: Wrong project =) And I think we can omit this altogether

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+For details, tutorials, and examples, please have a look at our `documentation`_.
+
+.. _`documentation`: https://luthaf.fr/rascaline/index.html
```
Comment (Luthaf) [inline README.rst:19]: wrong link

File: README.rst (hunk 2 above)
Comment (PicoCentauri) [reply README.rst]: Done.

**Comment by Luthaf** [comment]
Closing in favor of #91

**Pull request closed without merging**